File: backend/models/file_cluster.py
from bson import ObjectId
from datetime import datetime, timezone

def create_file_cluster(*, owner_id: ObjectId, name: str) -> dict:

    now = datetime.now(timezone.utc)

    return {

        "_id": ObjectId(),
        "ownerId": owner_id,
        "name": name,
        "createdAt": now,
        "isDeleted": False,

    }
    

File: backend/src/main/java/com/filespark/server/models/FileCluster.java
package com.filespark.server.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Document(collection = "file_clusters")
public class FileCluster {

    @Id
    private String id;

    private String ownerId;
    private String name;
    private Instant createdAt;
    private boolean isDeleted;

    public FileCluster(String ownerId, String name) {

        this.ownerId = ownerId;
        this.name = name;
        this.createdAt = Instant.now();
        this.isDeleted = false;

    }


    public String getId() {

        return id;

    }

    public String getOwnerId() {

        return ownerId;

    }

    public String getName() {

        return name;

    }

    public Instant getCreatedAt() {

        return createdAt;

    }

    public boolean isDeleted() {

        return isDeleted;

    }

}
